### PR TITLE
Update ConfigManagementPlugins handling

### PR DESCRIPTION
**Pull request by mhjacks**
1) Make configmanagement plugins optional
2) Allow for old style cmp's via legacyConfigManagementPlugins
3) All for "CMP 2.0" plugins aka sidecar plugins (now preferred)
4) Add new argoCD key to clusterGroup values with needed schema changes

### CODE REVIEW

**Comment by mhjacks** [comment]
Note: experimentation seems to indicate that CMP 2.0 deliberately makes it difficult (that is, practically impossible) to run executables inside the repository, so this feature will not support doing that with CMP 2.0. Users who really need that can used the CMP 1.0 framework (Which this PR supports) or plan to migrate.

File: clustergroup/templates/plumbing/argocd-cmp-plugin-cms.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{{- if len $.Values.clusterGroup.argoCD.configManagementPlugins }}
+{{- range $cmp := $.Values.clusterGroup.argoCD.configManagementPlugins }}
```
Comment (beekhof) [inline clustergroup/templates/plumbing/argocd-cmp-plugin-cms.yaml:2]: Is this if clause redundant here?

File: examples/kustomize-renderer/kustomize
Comment (beekhof) [inline examples/kustomize-renderer/kustomize:1]: If we don't test plugins, we also won't know if we break the templates we use to drive them.
Maybe we still need this example chart?

File: clustergroup/templates/plumbing/argocd-cmp-plugin-cms.yaml (hunk 1 above)
Comment (mhjacks) [reply clustergroup/templates/plumbing/argocd-cmp-plugin-cms.yaml]: Yes, it is. Fixed.

File: examples/kustomize-renderer/kustomize
Comment (mhjacks) [reply examples/kustomize-renderer/kustomize:1]: True. I re-added it.

File: clustergroup/templates/plumbing/argocd.yaml [hunk 2]
```diff
@@ -36,36 +36,9 @@ spec:
...
-            --set global.namespace=$ARGOCD_APP_NAMESPACE
-            --set global.pattern={{ .Values.global.pattern }}
-            --set global.clusterDomain={{ .Values.global.clusterDomain }}
-            --set global.hubClusterDomain={{ .Values.global.hubClusterDomain }}
-            --set global.localClusterDomain={{ coalesce .Values.global.localClusterDomain .Values.global.hubClusterDomain }}
-            --set clusterGroup.name={{ .Values.clusterGroup.name }}
-            --post-renderer ./kustomize"]
+  {{- if len $.Values.clusterGroup.argoCD.legacyConfigManagementPlugins }}
+  configManagementPlugins: | {{ $.Values.clusterGroup.argoCD.legacyConfigManagementPlugins | nindent 4 }}
+  {{- end }}
```
[the remark this answers is not in the document]
Comment (mhjacks) [reply clustergroup/templates/plumbing/argocd.yaml]: No, it's redundant. I coded this before I knew that the mechanism had been removed in current upstream. It's gone now.

**Pull request merged**